doc/examples/applications/plot_3d_structure_tensor.py

```diff
@@ -0,0 +1,160 @@
+
+#####################################################################
+# For the purposes of this tutorial, we shall consider only the second colour
+# channel, which leaves us with a 3D single-channel image. What is the range
+# of values?
+
+n_plane, n_Y, n_X, n_chan = data.shape
+v_min, v_max = data[:, :, :, 1].min(), data[:, :, :, 1].max()
+print(f'range: ({v_min}, {v_max})')
+
+#####################################################################
+# Let us visualize the middle slice of our 3D image.
+
+px.imshow(
+    data[n_plane // 2, :, :, 1],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'}
+)
+
+#####################################################################
+# Let us pick a specific region, which shows relative X-Y isotropy. In
+# contrast, the gradient is quite different (and, for that matter, weak) along
+# Z.
+
+sample = data[5:13, 380:410, 370:400, 1]
+step = 3
+cols = sample.shape[0] // step + 1
+_, axes = plt.subplots(nrows=1, ncols=cols, figsize=(16, 8))
+
+for it, (ax, image) in enumerate(zip(axes.flatten(), sample[::step])):
+    ax.imshow(image, cmap='gray', vmin=v_min, vmax=v_max)
+    ax.set_title(f'Plane = {5 + it * step}')
+    ax.set_xticks([])
+    ax.set_yticks([])
+
+#####################################################################
+# To view the sample data in 3D, run the following code:
+#
+# .. code-block:: python
+#
+#     import plotly.graph_objects as go
+#
+#     (n_Z, n_Y, n_X) = sample.shape
+#     Z, Y, X = np.mgrid[:n_Z, :n_Y, :n_X]
+#
+#     fig = go.Figure(
+#         data=go.Volume(
+#             x=X.flatten(),
+#             y=Y.flatten(),
+#             z=Z.flatten(),
+#             value=sample.flatten(),
+#             opacity=0.5,
+#             slices_z=dict(show=True, locations=[4])
+#         )
+#     )
+#     fig.show()
+
+#####################################################################
+# Compute structure tensor
+# ========================
+# About the brightest region (i.e., at X ~ 22 and Y ~ 17), we can see
+# variations (and, hence, strong gradients) along either X or Y over a typical
+# length of 3.
+
+px.imshow(
+    sample[0, :, :],
+    zmin=v_min,
+    zmax=v_max,
+    labels={'x': 'Y', 'y': 'X', 'color': 'intensity'},
+    title='Interactive view of bottom slice of sample data.'
+)
+
+#####################################################################
+# Therefore, we choose a 'width' of 3 for the window function.
+
+sigma = 3
+A_elems = feature.structure_tensor(sample, sigma=sigma)
+
+#####################################################################
+# We can then compute the eigenvalues of the structure tensor.
+
+eigen = feature.structure_tensor_eigenvalues(A_elems)
+eigen.shape
+
+#####################################################################
+# Where is the largest eigenvalue?
+
+coords = np.unravel_index(eigen.argmax(), eigen.shape)
+assert coords[0] == 0  # by definition
+coords
+
```

Comment:
I think it would be nice to also plot all 3 eigenvalues for a single slice (e.g. 1 subplot per eigenvalue, using uniform brightness scale across subplots).

Reply:
I was suggesting a figure something like this, which shows the spatial distribution of the eigenvalues. Do you think it is helpful?


```Python
fig, axes = plt.subplots(1, 3)
vmax = eigen[0][coords[1]].max()
axes[0].imshow(eigen[0][coords[1]], vmin=0, vmax=vmax, cmap=plt.cm.viridis)
axes[0].set_title('eigenvalue 0')
axes[1].imshow(eigen[1][coords[1]], vmin=0, vmax=vmax, cmap=plt.cm.viridis)
axes[1].set_title('eigenvalue 1')
axes[2].imshow(eigen[2][coords[1]], vmin=0, vmax=vmax, cmap=plt.cm.viridis)
axes[2].set_title('eigenvalue 2')
for ax in axes:
    ax.set_axis_off()
```
![evs](https://user-images.githubusercontent.com/6528957/105775217-7329f600-5f34-11eb-92a6-bcaf4156dc3e.png)

Reply:
I'm aware I hadn't addressed all of your review comments -- looking now! Thanks.

Reply:
Yes, I think it is helpful: It makes it easier to follow the rest of the story. I've added the plot in d82d754. I couldn't resist but use `facet_col` with plotly's imshow... :heart: @emmanuelle :wink: